src/urllib3/exceptions.pyi

```diff
@@ -0,0 +1,50 @@
+from typing import Any, Optional, Union, Tuple
+
+class HTTPError(Exception): ...
+class HTTPWarning(Warning): ...
+
+class PoolError(HTTPError):
+    pool: Any
+    def __init__(self, pool: Any, message: str) -> None: ...
```

Comment:
The `pool` here is from `connectionpool.py`, but importing here would create a circular dependency.

Reply:
If you're hitting circular imports you can sometimes use this construction:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from urllib3.connectionpool import ConnectionPool

class PoolError:
    pool: "ConnectionPool"
```